js/main.js

```diff
@@ -0,0 +1,113 @@
+var $h2GeneralSetup = document.getElementById('append-gen-joke');
+var $h2GeneralPunchline = document.getElementById('append-gen-punchline');
+
+var $h2ProgrammingSetup = document.getElementById('append-prog-joke');
+var $h2ProgrammingPunchline = document.getElementById('append-prog-punchline');
+
+function programmingJokes() {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://official-joke-api.appspot.com/jokes/programming/random');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var strJokeProg = JSON.stringify(xhr.response[0].setup);
+    var progJoke = document.createTextNode(strJokeProg);
+    $h2ProgrammingSetup.appendChild(progJoke);
+
+    var strPunchlineProg = JSON.stringify(xhr.response[0].punchline);
+    var progPunchline = document.createTextNode(strPunchlineProg);
+    $h2ProgrammingPunchline.appendChild(progPunchline);
+  });
+  xhr.send();
+}
+
+function generalJokes() {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://official-joke-api.appspot.com/random_joke');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var strJokeGeneral = JSON.stringify(xhr.response.setup);
```

Comment:
Is the stringify really necessary here? This looks like it is a string already 🤔